test/utils.py

```diff
@@ -234,21 +250,40 @@ def start(self, connection_test=True,
                 instance within three seconds deadline after port bound (an attempt to connect is
                 made if there is an option connection_test=True that is present by default).
         """
-        popen_obj = subprocess.Popen(["tarantool", self._instance_cfg_file_name], cwd=self._tmpdir)
-        file_with_port_path = str(self._tmpdir) + '/' + self._instance_cfg_file_name + '.port'
-
-        # Waiting 3 seconds for instance configure itself and dump bound port to temp file.
-        deadline = time.time() + 3
-        while time.time() < deadline:
-            if os.path.exists(file_with_port_path) and os.path.getsize(file_with_port_path) > 0:
-                break
-            time.sleep(0.1)
-        else:
-            raise RuntimeError('Could not find a file with an instance bound port or empty file')
+        tnt_start_cmd = [
+            "tarantool",
+            "--name",
+            instance_name,
+            "--config",
+            self._instance_cfg_file_name,
+        ]
+        major_version, _ = get_tarantool_version()
+        if major_version < 3 or use_lua:
+            tnt_start_cmd = ["tarantool", self._instance_cfg_file_name]
+        popen_obj = subprocess.Popen(tnt_start_cmd, cwd=self._tmpdir)
+
+        # Search for file with port only if port directly not provided.
+        if len(instance_port) == 0:
+            file_with_port_path = (
+                str(self._tmpdir) + "/" + self._instance_cfg_file_name + ".port"
+            )
+            # Waiting 3 seconds for instance configure itself and dump bound port to temp file.
+            deadline = time.time() + 3
+            while time.time() < deadline:
```

Comment:
NIT: util.wait_file can be used here, but if u also want to check size u will need to modify function/can leave as is.

Reply:
This code was before me. I just added condition if port was given not from file. So I leaved it as is.